log4j-nosql/src/main/java/org/apache/logging/log4j/nosql/appender/mongodb/MongoDbProvider.java

```diff
@@ -93,130 +103,249 @@ public String toString() {
+	                    if (Strings.isNotEmpty(databaseName)) {
+	                        database = ((MongoClient) object).getDB(databaseName);
+	                    } else {
+	                        LOGGER.error("The factory method [{}.{}()] returned a MongoClient so the database name is "
+	                                + "required.", factoryClassName, factoryMethodName);
+	                        return null;
+	                    }
+	                } else if (object == null) {
+	                    LOGGER.error("The factory method [{}.{}()] returned null.", factoryClassName, factoryMethodName);
+	                    return null;
+	                } else {
+	                    LOGGER.error("The factory method [{}.{}()] returned an unsupported type [{}].", factoryClassName,
+	                            factoryMethodName, object.getClass().getName());
+	                    return null;
+	                }
+
+	                description = "database=" + database.getName();
+	                final List<ServerAddress> addresses = database.getMongo().getAllAddress();
+	                if (addresses.size() == 1) {
+	                    description += ", server=" + addresses.get(0).getHost() + ", port=" + addresses.get(0).getPort();
+	                } else {
+	                    description += ", servers=[";
+	                    for (final ServerAddress address : addresses) {
+	                        description += " { " + address.getHost() + ", " + address.getPort() + " } ";
+	                    }
+	                    description += "]";
+	                }
+	            } catch (final ClassNotFoundException e) {
+	                LOGGER.error("The factory class [{}] could not be loaded.", factoryClassName, e);
+	                return null;
+	            } catch (final NoSuchMethodException e) {
+	                LOGGER.error("The factory class [{}] does not have a no-arg method named [{}].", factoryClassName,
+	                        factoryMethodName, e);
+	                return null;
+	            } catch (final Exception e) {
+	                LOGGER.error("The factory method [{}.{}()] could not be invoked.", factoryClassName, factoryMethodName,
+	                        e);
+	                return null;
+	            }
+	        } else if (Strings.isNotEmpty(databaseName)) {
+	            final List<MongoCredential> credentials = new ArrayList<>();
+	            description = "database=" + databaseName;
+	            if (Strings.isNotEmpty(userName) && Strings.isNotEmpty(password)) {
+	                description += ", username=" + userName + ", passwordHash="
+	                        + NameUtil.md5(password + MongoDbProvider.class.getName());
+	                credentials.add(MongoCredential.createCredential(userName, databaseName, password.toCharArray()));
+	            }
+	            try {
+	                final int portInt = TypeConverters.convert(port, int.class, DEFAULT_PORT);
```

Comment:
As mentioned elsewhere, if you make the field an int, you don't need to do this.